packages/imperative/src/profiles/src/utils/V1ProfileConversion.ts

```diff
@@ -0,0 +1,185 @@
+        V1ProfileConversion.crashInTeamConfigMode();
+
+        let meta: IMetaProfile<T>;
+        try {
+            meta = readYaml.load(fs.readFileSync(path), "utf8");
+        } catch (err) {
+            throw new ImperativeError({
+                msg: `Error reading profile file ("${path}"). Error Details: ${err.message}`,
+                additionalDetails: err
+            }, {tag: V1ProfileConversion.ERROR_ID});
+        }
+        return meta;
+    }
+
+    /**
+     * Accepts the profiles root directory and returns all directories within. The directories within the root
+     * directory are all assumed to be profile type directories (potentially containing a meta file and profiles
+     * of that type).
+     * @static
+     * @param {string} profileRootDirectory - The profiles root directory to obtain all profiles from.
+     * @returns {string[]} - The list of profiles returned or a blank array
+     * @memberof V1ProfileConversion
+     */
+    public static getAllProfileDirectories(profileRootDirectory: string): string[] {
+        V1ProfileConversion.crashInTeamConfigMode();
+
+        let names: string[] = [];
+        try {
+            names = fs.readdirSync(profileRootDirectory);
+            names = names.filter((name) => {
+                // only return directories, not files
+                const stats = fs.statSync(pathPackage.join(profileRootDirectory, name));
+                return stats.isDirectory();
+            });
+        } catch (e) {
+            throw new ImperativeError({
+                msg: `An error occurred attempting to read all profile directories from "${profileRootDirectory}". ` +
+                    `Error Details: ${e.message}`,
+                additionalDetails: e
+            }, {tag: V1ProfileConversion.ERROR_ID});
+        }
+        return names;
+    }
+
+    /**
+     * Accepts the profile directory location for a type, reads all filenames, and returns a list of
+     * profile names that are present within the directory (excluding the meta profile)
+     * @static
+     * @param {string} profileTypeDir - The directory for the type
+     * @param {string} ext - the extension for the profile files (normally YAML)
+     * @param {string} metaNameForType - the meta name for this type
+     * @returns {string[]} - A list of all profile names (without path/ext)
+     * @memberof V1ProfileConversion
+     */
+    public static getAllProfileNames(profileTypeDir: string, ext: string, metaNameForType: string): string[] {
+        V1ProfileConversion.crashInTeamConfigMode();
+
+        const names: string[] = [];
+        try {
+            let profileFiles = fs.readdirSync(profileTypeDir);
+            profileFiles = profileFiles.filter((file) => {
+                const fullFile = pathPackage.resolve(profileTypeDir, file);
+                const isYamlFile = fullFile.length > ext.length && fullFile.substring(
+                    fullFile.length - ext.length) === ext;
```

Comment:
This is a creative method way to verify the file extension 😋 
Out of curiosity, would this behave differently than `path.extname(fullFile) === ext`?

Reply:
Probably not. I do not really know. This code came from the original ProfileIO.ts file and has probably existed unmodified in this source since before I joined the project.